core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -493,7 +525,17 @@ public HTTPClient build() {
         interceptor = loadInterceptorDynamically(SIGV4_REQUEST_INTERCEPTOR_IMPL, properties);
       }
 
-      return new HTTPClient(uri, baseHeaders, mapper, interceptor, properties);
+      BasicCredentialsProvider credsProvider = null;
+
+      if (proxy != null && proxyUsername != null && proxyPassword != null) {
+        credsProvider = new BasicCredentialsProvider();
+        credsProvider.setCredentials(
+            new AuthScope(proxy),
+            new UsernamePasswordCredentials(proxyUsername, proxyPassword.toCharArray()));
```

Comment:
If we make the API change I mentioned here https://github.com/apache/iceberg/pull/10052/files#r1543172906, we would remove this